apollo-router/src/services/http/service.rs

```diff
@@ -267,6 +272,8 @@ impl tower::Service<HttpRequest> for HttpClientService {
             "net.transport" = "ip_tcp",
             //"apollo.subgraph.name" = %service_name,
             //"graphql.operation.name" = %operation_name,
+            "otel.status_code" = tracing::field::Empty,
+            "http.response.status_code" = tracing::field::Empty,
```

Comment:
This shouldn't be enabled by default it should follow the custom telemetry configuration like started [here](https://github.com/apollographql/router/pull/6125)

Reply:
Agreed, but there isn't a way to do that today, as I understand it. This span is created internally here inside the HTTP client service, so there wouldn't be a way for a plugin to add to it. Just above this, it similarly adds other attributes like `http.route` and `http.url`. If feel strongly that `http.response.status_code` shouldn't be here, I can remove it, but it's extremely helpful to be able to see this on the trace.

Reply:
In the interest of getting this landed, I've removed the changes to the HTTP client span. This will just set the `OTEL_STATUS_CODE` on the connector trace, making it visible in traces.

Reply:
I agree it's valuable but everything we add by default cannot be removed afterward because it would create a breaking change so I prefer to not add it right now. Thanks